hazelcast-client/src/main/java/com/hazelcast/client/proxy/NearCachedClientMapProxy.java

```diff
@@ -391,18 +392,29 @@ protected void loadAllInternal(boolean replaceExistingValues, Collection<?> keys
     }
 
     @Override
-    @SuppressWarnings("unchecked")
+    @SuppressWarnings({"checkstyle:npathcomplexity", "unchecked"})
     protected List<MapGetAllCodec.ResponseParameters> getAllInternal(Set<K> keys, Map<Integer, List<Data>> partitionToKeyData,
                                                                      Map<K, V> result) {
         Map<Object, Long> reservations = new HashMap<Object, Long>();
         try {
             if (serializeKeys) {
                 fillPartitionToKeyData(keys, partitionToKeyData);
+                boolean allEmpty = true;
```

Comment:
This is just a temporary fix to prevent the remote calls if all values are served from the Near Cache. A proper fix is made in https://github.com/hazelcast/hazelcast/pull/10714